src/goto-instrument/cover.cpp

```diff
@@ -251,14 +256,59 @@ std::unique_ptr<cover_configt> get_cover_config(
   return cover_config;
 }
 
+/// Build data structures controlling coverage from command-line options.
+/// Include options that depend on the main function specified by the user.
+/// \param options: command-line options
+/// \param main_id: symbol of the user-specified main program function
+/// \param symbol_table: global symbol table
+/// \param message_handler: used to log incorrect option specifications
+/// \return a cover_configt on success, or null otherwise.
+std::unique_ptr<cover_configt> get_cover_config(
+  const optionst &options,
+  const irep_idt &main_id,
+  const symbol_tablet &symbol_table,
+  message_handlert &message_handler)
+{
+  messaget msg(message_handler);
+  std::unique_ptr<cover_configt> cover_config =
+    get_cover_config(options, symbol_table, message_handler);
+
+  if(!cover_config)
+    return {};
+
+  std::string cover_only = options.get_option("cover-only");
+  const symbolt main_symbol = symbol_table.lookup_ref(main_id);
+
```

Comment:
only do the lookup if needed, i.e., in the "function" case. Also, use a reference, don't copy the symbol.

Reply:
It is used in two if branches (function and file), but ok, I'll put the lookup in each.